From: Markus Koschany <apo@gambaru.de>
Date: Thu, 25 Jul 2013 08:19:16 +0200
Subject: kFreeBSD ENODATA

         Fix a FTBFS on kFreeBSD with libusb 1.0. ENODATA is
         undeclared on kfreeBSD thus use the equivalent ENOATTR.

Forwarded: gphoto-devel@lists.sourceforge.net
Reviewed-by: Herbert Parentes Fortes Neto <hpfn@ig.com.br>
Last-Update: 2016-03-26

Index: libgphoto2-2.5.10/libgphoto2_port/libusb1/libusb1.c
===================================================================
--- libgphoto2-2.5.10.orig/libgphoto2_port/libusb1/libusb1.c
+++ libgphoto2-2.5.10/libgphoto2_port/libusb1/libusb1.c
@@ -23,6 +23,13 @@
 #include "config.h"
 #include <gphoto2/gphoto2-port-library.h>
 
+/* Workaround for GNU/kFreeBSD */
+#if defined(__FreeBSD_kernel__)
+#ifndef ENODATA
+#define ENODATA ENOATTR
+#endif
+#endif
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <fcntl.h>
